### PR TITLE
Fixed Issues #118 and #119 and added support for excludeSubkeys

### PR DESCRIPTION
**Pull request by TimLudwinski**
Ok.  I finally finished successfully running my modified version in our very large database.  (Our database has 182 million records and is 1.3 TB so this took a long time).

### CODE REVIEW

**Comment by TimLudwinski** [comment]
Sorry, I'll have to fix whatever bug is occurring here next week.

**Comment by JamesCropcho** [comment]
Adding semantic reference to #118 and #119

**Comment by TimLudwinski** [comment]
Fixed failing test.  Please merge.

File: README.markdown [hunk 1]
```diff
@@ -153,6 +153,29 @@ Sometimes you want to see the keys and types come in as it happens.  Maybe you h
 
     $ mongo test --eval "var collection = 'users', sort = { updated_at : -1 }, logKeysContinuously = true" variety.js
 
+#### Exclude Subkeys ####
+Sometimes you inherit a database full of junk.  Maybe the previous developer put data in the database keys, which causes variety to go out of memory when run.  After you've run the `logKeysContinuously` to figure out which subkeys may be a problem, you can use this option to run variety without those subkeys.
```
Comment (JamesCropcho) [inline README.markdown]: "Variety" as in "the schema analyzer" is capitalized.

File: README.markdown [hunk 2]
```diff
@@ -177,6 +200,11 @@ To persist to an alternate MongoDB database, you may specify the following param
 $ mongo test --quiet --eval "var collection = 'users', persistResults=true, resultsDatabase='db.example.com/variety' variety.js
 ```
 
+### Reserved Keys ###
+Variety expects keys to be well formed, not having any '.'s in them (mongo 2.4 allows dots in certain cases).  Also mongo uses the psudo keys 'XX' and keys coresponding to the regex 'XX\d+XX.*' for use with arrays.  You can change the string XX in these patterns to whatever you like if there is a conflict in your database using the `arrayEscape` parameter.
```
Comment (JamesCropcho) [inline README.markdown]: "pseudo" is misspelled.

**Comment by todvora** [comment]
One more thing Tim - would it be please possible to write a test for the *excludeSubkeys* feature? Or should I do it? It shouldn't be much different from [LimitResultsAnalysis](https://github.com/variety/variety/blob/master/spec/LimitResultsAnalysisTest.js) test, for example. 

Thanks!
Tomas

**Comment by TimLudwinski** [comment]
I won't be able to get to the unit tests for the next few weeks, since I'm busy.

**Comment by todvora** [comment]
There haven't been any more objections, so I took all the @TimLudwinski's changes, added some tests and merged everything together. The features are well documented, code is clean, [tests are passing](https://travis-ci.org/variety/variety). Thanks everyone for your work!

Regards,
Tomas

**Comment by JamesCropcho** [comment]
Awesome, man ( @todvora ). Thank you, too for your work on this!

**Pull request merged**